src/components/Modals/NewCourse/CourseSelector.vue

```diff
@@ -25,35 +25,44 @@
 
 <script lang="ts">
 import { PropType, defineComponent } from 'vue';
+import Fuse from 'fuse.js';
 import { fullCoursesArray } from '@/assets/courses/typed-full-courses';
 
 const getMatchingCourses = (
   searchText: string,
   filter?: (course: CornellCourseRosterCourse) => boolean
 ): readonly CornellCourseRosterCourse[] => {
-  // search after value length of 2 to reduce search times of courses
-  if (!searchText || searchText.length < 2) return [];
-  /* code array for results that contain course code and title array for results that contain title */
-  const code: CornellCourseRosterCourse[] = [];
-  const title: CornellCourseRosterCourse[] = [];
-
-  const filteredCourses = filter != null ? fullCoursesArray.filter(filter) : fullCoursesArray;
-  for (const course of filteredCourses) {
-    const courseCode = `${course.subject} ${course.catalogNbr}`;
-    if (courseCode.toUpperCase().includes(searchText)) {
-      code.push(course);
-    } else if (course.titleLong.toUpperCase().includes(searchText)) {
-      title.push(course);
-    }
+  interface SearchableCourse extends CornellCourseRosterCourse {
+    courseCode?: string;
+    fullCourseString?: string;
   }
 
-  // Sort both results by title
-  code.sort((first, second) => first.titleLong.localeCompare(second.titleLong));
-  title.sort((first, second) => first.titleLong.localeCompare(second.titleLong));
+  const courses: readonly SearchableCourse[] =
+    filter != null ? fullCoursesArray.filter(filter) : fullCoursesArray;
+  courses.map((course: SearchableCourse) => {
+    course.courseCode = `${course.subject} ${course.catalogNbr}`;
+    course.fullCourseString = `${course.subject} ${course.catalogNbr} ${course.titleLong}`;
+    return course;
+  });
+  const options = {
+    keys: [
+      { name: 'courseCode', weight: 2 },
```

Comment:
Probably not needed since it's already covered by subject and catalogNbr?

Reply:
Agreed. Could the same be also said for fullCourseString if we can derive it from subject, catalogNbr, and titleLong? Did you create a new field because there isn't another way of weighting them all together?

Reply:
It's funky. Without the fullCourseString it doesn't match substrings of the entry with each property. I added all these things to get it to act the way I want it to. But, after thinking about it, it might just be better to create an array of strings of only fullCourseString and strictly match to that. I think that's the most simple implementation